noir_stdlib/src/merkle.nr

```diff
@@ -5,16 +5,15 @@
 // and the hashpath proves this
 // Currently we assume that it is a binary tree, so depth k implies a width of 2^k
 // XXX: In the future we can add an arity parameter
-#[foreign(merkle_membership)]
-fn check_membership(_root : Field, _leaf : Field, _index : Field, _hash_path: [Field]) -> Field {}
-
-
-#[alternative(merkle_membership)]
-fn check_membership_in_noir(root : Field, leaf : Field, index : Field, hash_path: [Field]) -> Field {
-    (compute_root_from_leaf(leaf, index, hash_path) == root) as Field
+fn check_membership(_root : Field, _leaf : Field, _index : Field, _hash_path: [Field]) -> Field {
+    (compute_merkle_root(_leaf, _index, _hash_path) == _root) as Field
 }
 
+#[foreign(compute_merkle_root)]
+fn compute_merkle_root(_leaf : Field, _index : Field, _hash_path: [Field]) -> Field {}
+
 // Returns the root of the tree from the provided leaf and its hashpath, using pedersen hash
+#[alternative(compute_merkle_root)]
 fn compute_root_from_leaf(leaf : Field, index : Field, hash_path: [Field]) -> Field {
```

Comment:
Maybe we rename this to `compute_merkle_root_in_noir` or `compute_merkle_root_native`? I feel like the names `compute_root_from_leaf` and `compute_merkle_root` aren't distinct enough. However, this is not blocking in my opinion and could be done in a small PR in the next release

Reply:
I left this as is to avoid a breaking change to the stdlib (same reason that `check_membership` still returns a `Field`).

(I'm not counting the removal of `check_membership_in_noir` as breaking as it should only be called through the fallback mechanism and not directly)

Reply:
I see. I'm good with leaving this as is right now then and renaming in a separate PR